Source/AMRInterpolator/SphericalExtraction.hpp

```diff
@@ -7,67 +7,72 @@
 #define SPHERICALEXTRACTION_HPP_
 
 #include "SphericalGeometry.hpp"
+#include "SphericalGeometryUniform.hpp"
 #include "SphericalHarmonics.hpp"
 #include "SurfaceExtraction.hpp"
 
+struct spherical_extraction_params_t : surface_extraction_params_t
+{
+    int &num_extraction_radii = num_surfaces;
+    std::vector<double> &extraction_radii = surface_param_values;
+    int &num_points_theta = num_points_u;
+    int &num_points_phi = num_points_v;
+    std::array<double, CH_SPACEDIM> center; //!< the center of the spherical
+                                            //!< shells
+    std::array<double, CH_SPACEDIM> &extraction_center = center;
+    int num_modes;                          //!< the number of modes to extract
+    std::vector<std::pair<int, int>> modes; //!< the modes to extract
+                                            //!< l = first, m = second
+    // constructor
+    spherical_extraction_params_t() = default;
+
+    // copy constructor defined due to references pointing to the wrong
+    // things with the default copy constructor
+    spherical_extraction_params_t(const spherical_extraction_params_t &params)
+        : surface_extraction_params_t(params), center(params.center),
+          num_modes(params.num_modes), modes(params.modes)
+    {
+    }
+};
+
 //! A child class of SurfaceExtraction for extraction on spherical shells
-class SphericalExtraction : public SurfaceExtraction<SphericalGeometry>
+template <typename spherical_geometry>
+class ModeExtraction : public SurfaceExtraction<spherical_geometry>
```

Comment:
Not sure if I like the name `ModeExtraction` as it's specific for spherical geometries. What about `SphericalModeExtraction`?